src/lib.rs

```diff
@@ -408,18 +444,20 @@ impl Socket {
     /// Consume the Socket and return the raw socket pointer.
     ///
     /// Failure to close the raw socket manually or call `from_raw`
-    /// will lead to a memory leak.
+    /// will lead to a memory leak. Also note that is function
+    /// relinquishes the reference on the context is was created from.
     pub fn into_raw(mut self) -> *mut c_void {
-        self.owned = false;
+        self.context = None;
```

Comment:
The latest push should fix that, including a (superficial, as promised ;-) test.